db/helpers/botUtils.js

```diff
@@ -45,16 +45,23 @@ function arrayHasValidParameters(arr) {
   if (Array.isArray(arr)) {
     for (let i = 0; i < arr.length; i++) {
       if ((typeof arr[i] === 'object') &&
-        (arr[i].hasOwnProperty('name')) && (arr[i].hasOwnProperty('type'))) {
-        if ((!constants.nameRegex.test(arr[i].name)) ||
+        (arr[i].hasOwnProperty('name')) &&
+        ((arr[i].hasOwnProperty('type') || arr[i].hasOwnProperty('value')))) {
+        if (!constants.nameRegex.test(arr[i].name)) {
+          throw new ValidationError({
+            message: 'Missing a valid name',
+          });
+        }
+
+        if ((arr[i].hasOwnProperty('type')) &&
```

Comment:
please consider adding one model at a time each in separate PR, then update with the associations after ... big PRs like this are VERY hard to review